task/apply-tags/0.2/migrations/0.2.sh

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Created for task: apply-tags@0.2
+# Creation time: 2025-06-05T10:43:11+00:00
+
+declare -r pipeline_file=${1:?missing pipeline file}
+
+# Determine the correct image-digest and image-url values based on task presence
+if yq -e '.spec.tasks[] | select(.name == "build-oci-artifact")' "$pipeline_file" >/dev/null; then
+    image_digest_value="\$(tasks.build-oci-artifact.results.IMAGE_DIGEST)"
+    image_url_value="\$(tasks.build-oci-artifact.results.IMAGE_URL)"
+elif yq -e '.spec.tasks[] | select(.name == "build-image-index")' "$pipeline_file" >/dev/null; then
+    image_digest_value="\$(tasks.build-image-index.results.IMAGE_DIGEST)"
+    image_url_value="\$(tasks.build-image-index.results.IMAGE_URL)"
+else
+    echo "Neither build-oci-artifact nor build-image-index tasks found."
+    exit 0
```

Comment:
One reason why the migration could fail to apply would be if users created their Components a long time ago, before even the single-arch pipeline started using the build-image-index task.

We can add handling for that (handle the `build-container` as fallback) or just document the necessary changes in the MIGRATION.md

Reply:
Added build-container task as well to the script + updated the MIGRATION.md as requested in different thread